src/index.ts

```diff
@@ -1,10 +1,11 @@
 import { connect, ErrorCode, JSONCodec } from "nats";
 import express, { json } from "express";
-
-const jsonCodec = JSONCodec();
+import { authorizationMiddleware } from "./lib/authorization.middleware";
```

Comment:
+1 for extracting each middleware in its own file.

2 things:
1. for consistency, we should keep camelCase for naming our files, that's what we're doing in all our repositories in this github org
2. I would recommend creating a `./middlewares` folder.
So we should have a `middlewares/authorization.ts` file instead